### DISCUSSION TITLE
Image under the Navigationbar & StatusBar

### DISCUSSION THREAD

**Discussion by salisuabubakar** [grahammendick/navigation#685] [Q&A] [answered]
Hello,

Thanks once again for your time. I would like to know if i can achieve the image below with The Navigation Router on Both android and ios. Thanks

<img width="283" alt="Screenshot 2023-03-03 at 3 36 11 PM" src="https://user-images.githubusercontent.com/6886556/222763738-159f9d2a-6463-4803-8f7d-45ef54137a6f.png">

**Comment by grahammendick** [comment]
You mean edge 2 edge, right?

On iOS, that's the default. So **don't** use a `SafeAreaView` and **don't** set `contentInsetAdjustmentBehavior` to 'automatic' on ScrollView/FlatList and the content will go under the status bar and notch. Also set the `barTintColor` of the `NavigationBar` to 'transparent' so the content shows through.

On Android, you need to [turn on some native settings](https://developer.android.com/develop/ui/views/layout/edge-to-edge). I don't think you need the Navigation router to do this for you but I'm pretty sure it would be easier if it did. I plan to look into this soon. If you're interested in contributing to the Navigation router then it might be something you could investigate? No pressure, of course.

**Comment by salisuabubakar** [comment]
Hi @grahammendick , 
I have got it working on ios. Yet to try it on android. I would like to say thank you once again. However, i couldnt get the transluscent  to work. Can i use The Navigation Router to achieve that . Thanks

**Reply by grahammendick** [reply]
On Android, like on iOS, you set the `barTintColor` of the `NavigationBar` to 'transparent'. Then you add negative `marginTop` of -60 to the `ScrollView` content, for example, so that it starts at the top of the screen.

**Reply by grahammendick** [reply]
In summary, this works on both iOS and Android to give a transparent navigation bar with the content underneath
```jsx
<View collapsable={false} style={{flex: 1}}>
  <NavigationBar barTintColor="transparent" />
  <ScrollView style={{ marginTop: Platform.OS === 'android' ? -60 : undefined}}>
</View>
```

**Reply by salisuabubakar** [reply]
It works on ios perfectly but on ondroid i still get empty space at the top. What can i do with the egde to edge in android to get it working.

**Reply by grahammendick** [reply]
I think you just need to turn [on edge to edge display](https://developer.android.com/develop/ui/views/layout/edge-to-edge#lay-out-in-full-screen) and then set the `barTintColor` prop of the `StatusBar` to 'transparent'

**Reply by grahammendick** [reply]
@salisuabubakar How you getting on?

**Reply by salisuabubakar** [reply]
Hi, on IOS it’s default like you said. When I tried it on android the negative you added on the scroll view pushes the navigationbar underneath the statusbar. 

Please help me get it done. Thanks for your time.

**Reply by salisuabubakar** [reply]
Hi, still waiting for your response. Thanks

**Reply by grahammendick** [reply]
Hey, you only updated me yesterday. It's on my list

**Reply by grahammendick** [reply]
I got it working on Android. I turned on [edge to edge](https://developer.android.com/develop/ui/views/layout/edge-to-edge#lay-out-in-full-screen) in the Activity
```java
@Override
protected void onCreate(Bundle savedInstanceState) {
  super.onCreate(savedInstanceState);
  WindowCompat.setDecorFitsSystemWindows(getWindow(), false);
}
```

Then I made the `NavigationBar` and `StatusBar` transparent and shifted the content up with -60 margin top.
```jsx
<View collapsable={false} style={{flex: 1}}>
  <NavigationBar title="Home" barTintColor="transparent">
    <StatusBar barTintColor="transparent" tintStyle="dark" />
  </NavigationBar>
  <ScrollView style={{ marginTop: -60}}>
</View>
```

**Reply by salisuabubakar** [reply]
Hi @grahammendick ,
Yes it works but the NavigationBar will be hidden.  On ios you will see the navigation bar with the back icon. I hope you got what am trying to say. Thanks

**Reply by grahammendick** [reply]
Give the `ScrollView` content a 'zIndex' of -1  so the `NavigationBar` appears on top

**Comment by grahammendick** [comment, marked as answer]
I released navigation-react-native v9.25.0 with edge-to-edge support.

To [enable edge-to-edge](https://developer.android.com/develop/ui/views/layout/edge-to-edge#enable-edge-to-edge-display) call [enableEdgeToEdge from within the MainActivity](https://developer.android.com/reference/androidx/activity/ComponentActivity#(androidx.activity.ComponentActivity).enableEdgeToEdge(androidx.activity.SystemBarStyle,androidx.activity.SystemBarStyle))

```kt
override fun onCreate(savedInstanceState: Bundle?) {
    enableEdgeToEdge();
    super.onCreate(savedInstanceState)
}
```
